services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -148,6 +149,22 @@ open class HankeServiceImpl(
         return savedHanke
     }
 
+    /**
+     * Create application when no existing hanke. Autogenerates hanke and applies application to it.
+     */
+    @Transactional
+    override fun generateHankeWithApplication(
+        application: Application,
+        userId: String
+    ): HankeWithApplications {
+        val hanke = generateHankeFrom(application)
+        val tunnus = hanke.hankeTunnus ?: throw HankeArgumentException("Hanke must have tunnus")
+        val createdApplication =
+            applicationService.create(application.withHanketunnus(tunnus), userId)
```

Comment:
Since Application is a data class, copy can be used instead of a custom method.
```suggestion
            applicationService.create(application.copy(hankeTunnus = tunnus), userId)
```

Reply:
This is now irrelevant as the implementation has changed.